sklearn/metrics/tests/test_ranking.py

```diff
@@ -881,6 +871,21 @@ def test_binary_clf_curve_implicit_pos_label(curve_func):
         np.testing.assert_allclose(int_curve_part, float_curve_part)
 
 
+@pytest.mark.parametrize("curve_func", [precision_recall_curve, roc_curve])
+@pytest.mark.parametrize("labels_type", ["list", "array"])
+def test_binary_clf_curve_implicit_bytes_pos_label(curve_func, labels_type):
+    # Check that using bytes class labels raises an informative
+    # error for any supported string dtype:
+    labels = _convert_container([b"a", b"b"], labels_type)
+    msg = (
+        "y_true takes value in {b'a', b'b'} and pos_label is not "
+        "specified: either make y_true take value in {0, 1} or "
+        "{-1, 1} or pass pos_label explicitly."
+    )
+    with pytest.raises(ValueError, match=msg):
+        curve_func(labels, [0.0, 1.0])
```

Comment:
For the CI to pass, we'll need to make sure the warning is raised and the previous error is raised. It's a little strange, but we can do it as follows:

```suggestion
    with warnings.catch_warnings(record=True) as record:
        with pytest.raises(ValueError, match=msg):
            curve_func(labels, [0.0, 1.0])

    assert str(record[0].message).startswith(
        "Support for labels represented as bytes is deprecated"
    )
```